src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4821,6 +4822,20 @@ jint os::init_2(void) {
     }
   }
 
+  // Check the availability of MADV_POPULATE_WRITE.
+  if (FLAG_IS_DEFAULT(UseMadvPopulateWrite) && UseMadvPopulateWrite) {
+    // Some downstream kernels recognize MADV_POPULATE_WRITE_value as another
+    // advice, so the check of versions is required here.
+    // See https://github.com/oracle/linux-uek/issues/23
+    if (Linux::_release_major < 5 ||
+        (Linux::_release_major == 5 && Linux::_release_minor < 14)) {
+      FLAG_SET_DEFAULT(UseMadvPopulateWrite, false);
+    } else {
+      FLAG_SET_DEFAULT(UseMadvPopulateWrite, (::madvise(0, 0, MADV_POPULATE_WRITE) == 0));
+    }
+  }
+  log_debug(gc, os)("UseMadvPopulateWrite=%d", (int)(UseMadvPopulateWrite ? 1 : 0));
```

Comment:
Is this temporary debugging code? We have -XX:+PrintFlagsFinal to check the resulting flag values.